packages/design-systems/src/components/composite/Datagrid/Datagrid.tsx

```diff
@@ -109,20 +111,23 @@ export const DataGrid = <TData extends Record<string, unknown>>(
   };
 
   useEffect(() => {
-    //Get header titles
-    const columnHeaders: Column<TData>[] = [];
-    table.getHeaderGroups().map((headerGroup) => {
-      headerGroup.headers.map((header) => {
-        columnHeaders.push({
-          //This is temporary structure, we will change this logic in coming days as required
-          label: header.column.columnDef.header as string,
-          value: header.column.columnDef.header as string,
-          meta: header.column.columnDef.meta as ColumnMetaWithTypeInfo<TData>,
-          disabled: false,
-        });
-      });
-    });
-    setColumnHeaders(columnHeaders);
+    //Get header titles from table columns
+    const cusotmFilterFields: Column<TData>[] = table.columns
+      .map((column) => {
+        if ("accessorKey" in column) {
+          return {
+            //This is temporary structure, we will change this logic in coming days as required
+            label: column.header as string,
+            value: column.header as string,
+            accessorKey: column.accessorKey as string,
+            disabled: false,
+            meta: column.meta,
+          };
+        }
+      })
+      .filter(Boolean) as Column<TData>[];
```

Comment:
ここでas使うくらいであればflatMap使ってもらいたいです

```suggestion
      .flatMap((column) => {
        if ("accessorKey" in column) {
          return [
            {
              //This is temporary structure, we will change this logic in coming days as required
              label: column.header as string,
              value: column.header as string,
              accessorKey: column.accessorKey as string,
              disabled: false,
              meta: column.meta,
            }
          ];
        } else {
          return []
        }
      })
```